src/services/codefixes/helpers.ts

```diff
@@ -592,7 +591,7 @@ function createTypeParameterName(index: number) {
 /** @internal */
 export function typeToAutoImportableTypeNode(checker: TypeChecker, importAdder: ImportAdder, type: Type, contextNode: Node | undefined, scriptTarget: ScriptTarget, flags?: NodeBuilderFlags, tracker?: SymbolTracker): TypeNode | undefined {
     let typeNode = checker.typeToTypeNode(type, contextNode, flags, tracker);
-    if (typeNode && isImportTypeNode(typeNode)) {
+    if (typeNode) {

```

Comment:
Not sure why that check was there in the first place. `tryGetAutoImportableReferenceFromTypeNode` is capable of taking any type node, and it will walk the type node in depth and replace all import types with simple type names and return an array of symbols to be imported. This check meant that a type like `Promise<import ('some import').T>` would be left as is because the import type was not directly the top type. While a type like `import('some import').G<import ('some import').T>` would be simplified to `G<T>` because the top level type was an import type.